libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyFile.java

```diff
@@ -249,28 +209,15 @@ private boolean init(URL policy, PolicyInfo newInfo) {
             return false;
         }
 
-        try (InputStreamReader isr = getInputStreamReader(PolicyUtil.getInputStream(policy))) {
+        try (InputStreamReader isr = getInputStreamReader(getInputStream(policy))) {
 
             PolicyParser pp = new PolicyParser(expandProperties);
             pp.read(isr);
 
-            KeyStore keyStore = null;
-            try {
-                keyStore = PolicyUtil.getKeyStore(
-                    policy,
-                    pp.getKeyStoreUrl(),
-                    pp.getKeyStoreType(),
-                    pp.getKeyStoreProvider(),
-                    pp.getStorePassURL()
-                );
-            } catch (Exception e) {
-                // ignore, treat it like we have no keystore
-            }
-
             Enumeration<PolicyParser.GrantEntry> enum_ = pp.grantElements();
             while (enum_.hasMoreElements()) {
                 PolicyParser.GrantEntry ge = enum_.nextElement();
-                addGrantEntry(ge, keyStore, newInfo);
```

Comment:
In addition to the keystore, it looks like the GrantEntry itself supports some additional configurations, like [signedBy](https://github.com/opensearch-project/OpenSearch/blob/main/libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyParser.java#L703-L705), that are unnecessary. I believe that signedBy is also used in the context of keystore, where you can specify that jars are signed by the alias of a particular key.

See [jarsigner](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/jarsigner.html) which is a tool shipped with the JDK for signing jars

Reply:
Similarly to signedBy, `principal` also goes unused within the project: https://docs.oracle.com/javase/7/docs/technotes/guides/security/PolicyFiles.html

Reply:
>In addition to the keystore, it looks like the GrantEntry itself supports some additional configurations, like [signedBy](https://github.com/opensearch-project/OpenSearch/blob/main/libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyParser.java#L703-L705), that are unnecessary.

You are right. I think I removed the traces of ```signedBy```, i will check if they are still out there.

Reply:
> I just realized that OpenSearch already has some custom logic around the policy files [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/bootstrap/Security.java#L232-L293).

Ah great fine. We can try to reuse that code part, over skimming, it does look sufficient to me.

Reply:
We need this code block since it sets the system properties needed for the PolicyParser to do property expansion.

Reply:
ack.

Reply:
the code block is not getting removed, i have just stripped off the ```keystore``` param.